ipld/hamt/src/node.rs

```diff
@@ -389,10 +389,11 @@ where
                 let deleted = child_node.rm_value(hashed_key, bit_width, depth + 1, key, store)?;
                 if deleted.is_some() {
                     *child = Pointer::Dirty(std::mem::take(child_node));
+
+                    // Clean to retrieve canonical form
+                    child.clean()?;
```

Comment:
Should this be backported to `ref-fvm`? If I understand correctly, they have the same issue on their side so calibnet can't work with any implementation using ref-fvm, right? https://github.com/filecoin-project/ref-fvm/blob/lemmih/forest/ipld/hamt/src/node.rs#L396

Reply:
Good question, I'm going to say yes. I'll open a PR on their side with the same fix and see what they say

Reply:
PR opened here: https://github.com/filecoin-project/ref-fvm/pull/343

Reply:
Thank you!